src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -1364,6 +1364,16 @@ bool StorageReplicatedMergeTree::tryExecuteMerge(const LogEntry & entry)
         return false;
     }
 
+    /// In some use cases merging can be more expensive than fetching
+    /// and it may be better to spread merges tasks across the replicas
+    /// instead of doing exactly the same merge cluster-wise
+    auto replica_to_execute_merge = merge_strategy_picker.pickReplicaToExecuteMerge(entry);
+    if (replica_to_execute_merge)
+    {
+        LOG_DEBUG(log, "Prefer fetching part {} from replica {} due execute_merges_on_single_replica_time_threshold", entry.new_part_name, replica_to_execute_merge.value());
```

Comment:
How do we guarantee that the same replica will be chosen in `findReplicaHavingPart`?

Reply:
Hash(Zookeeper path + target part name) % active_replicas_count

When replica added / removed the active_replicas_count can change, and there are no strong guarantees.